.github/workflows/update-code-injection-models.yml

```diff
@@ -0,0 +1,73 @@
+name: 'Update code injection models'
+
+on:
+  # Run weekly (at arbitrary time)
+  schedule:
+    - cron: '31 7 * * 5'
+  workflow_dispatch:
+
+permissions: {}
+
+jobs:
+  update-models:
+    permissions:
+      contents: write
+      pull-requests: write
+    name: Update models
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4
+        with:
+          node-version: 22
+
+      # Use custom working dir for downloading npm packages and cloning the CodeQL repo
+      - name: Create working dir
+        run: |
+          mkdir codeql-models-working-dir
+
+      - name: Install dependencies
+        run: |
+          npm install yaml@2.7.1 --no-save
+        working-directory: codeql-models-working-dir
+
+      - name: Check out CodeQL
+        id: checkout_codeql
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          repository: github/codeql
+          path: codeql-models-working-dir/codeql
+          sparse-checkout: |
+            actions/ql/lib/ext
+
+      - name: Update model file
+        run: node ../.github/workflows/process-codeql-sink-models.js
+        working-directory: codeql-models-working-dir
+
+      # If there are no changes, this action will not create a pull request
+      - name: Create or update pull request
+        uses: peter-evans/create-pull-request@271a8d0340265f705b14b6d32b9829c1cb33d45e # v7
+        with:
+          add-paths: |
+            code-injection-models.txt
```

Comment:
Have to explicitly specify this because the `codeql-models-working-dir` is also within the repository working tree.

Maybe that is not ideal because https://github.com/peter-evans/create-pull-request/tree/v7.0.8?tab=readme-ov-file#add-specific-paths mentions that other files (i.e. `codeql-models-working-dir`?) will be stashed and restored.

Might be better to have `codeql-models-working-dir` outside the working tree, maybe can created it as `../codeql-models-working-dir`?